lib/internal/process/promises.js

```diff
@@ -38,6 +38,15 @@ const {
 } = require('internal/async_hooks');
 const { isErrorStackTraceLimitWritable } = require('internal/errors');
 
+const { AsyncContextFrame } = require('internal/async_context_frame');
+
+let hasAsyncContextFrame_;
+function hasAsyncContextFrame() {
```

Comment:
Is there a significant performance improvement in copying this into several files?

Reply:
Probably not. Could be moved into internal/async_context_frame. It was just a direct conversion of the previous code which checked by existence of the AsyncContextFrame export of the native binding.